### PR TITLE
[BUG] Failed to update table status on concurrent compaction and data load.

### PR DESCRIPTION
**Pull request by ravikiran23**
If the table status file needs to be updated then the table status lock should be acquired and done.
if the locking is failed because of other process locking then need to fail the complete Process.
If table status lock is failed to acquire then complete data load or the compaction should fail.
Correcting the local lock.
Adding the FT for compaction locking.

### CODE REVIEW

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java [hunk 1]
```diff
@@ -687,51 +690,78 @@ public static String getLoadFolderPath(CarbonLoadModel loadModel, String cubeNam
    * This API will write the load level metadata for the loadmanagement module inorder to
    * manage the load and query execution management smoothly.
    */
```
Comment (gvramana) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: add comment for return type

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java [hunk 2]
```diff
@@ -687,51 +690,78 @@ public static String getLoadFolderPath(CarbonLoadModel loadModel, String cubeNam
...
-    loadMetadataDetails.setLoadName(String.valueOf(loadCount));
-    loadMetadataDetails.setLoadStartTime(startLoadTime);
-    LoadMetadataDetails[] listOfLoadFolderDetailsArray = null;
+    boolean status = false;
+
+    String metaDataFilepath =
+        loadModel.getCarbonDataLoadSchema().getCarbonTable().getMetaDataFilepath();
+
+    ICarbonLock carbonLock =
+        CarbonLockFactory.getCarbonLockObj(metaDataFilepath, LockUsage.TABLE_STATUS_LOCK);
```
Comment (gvramana) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: Add function getLock in status manager, which will create lockobject and give to user of status manager

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 3]
```diff
@@ -791,12 +791,18 @@ object CarbonDataRDDFactory extends Logging {
...
+          val status = CarbonLoaderUtil
             .recordLoadMetadata(currentLoadCount,
               metadataDetails,
               carbonLoadModel,
               loadStatus,
-              loadStartTime)
+              loadStartTime
+            )
+          if (!status) {
+            val message = "Dataload failure"
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: Give clear message why load has failed.

Comment (ravikiran23) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java (hunk 1 above)
Comment (ravikiran23) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: fixed

File: integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java (hunk 2 above)
Comment (ravikiran23) [inline integration/spark/src/main/java/org/carbondata/spark/load/CarbonLoaderUtil.java]: fixed

**Comment by gvramana** [comment]
LGTM

**Pull request merged**